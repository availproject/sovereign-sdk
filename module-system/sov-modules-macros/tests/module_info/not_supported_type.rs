use sov_modules_api::{Context, ModuleInfo};
use sov_state::StateMap;

#[derive(ModuleInfo)]
struct TestStruct<C: Context> {
    #[address]
    address: C::Address,

    #[state]
    test_state1: [usize; 22],

    #[state]
    test_state2: StateMap<u32, u32>,
}

fn main() {}
