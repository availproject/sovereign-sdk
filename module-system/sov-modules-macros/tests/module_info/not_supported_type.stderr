error: Type not supported by the `ModuleInfo` macro
  --> tests/module_info/not_supported_type.rs:10:18
   |
10 |     test_state1: [usize; 22],
   |                  ^^^^^^^^^^^
